src/pages/InitMsig.vue

```diff
@@ -214,63 +229,77 @@ export default defineComponent({
 			errorMessage(this.key);
 			openErrorNotificationWithIcon(error.message);
 		},
+		createMultisig() {
+			let sender = algosdk.decodeUnsignedTransaction(
+				convertBase64ToUint8Array(this.unsignedInput)
+			);
+			let fromAddr = algosdk.encodeAddress(sender.from.publicKey);
+			let addressList = this.addresses.map((x) => {
+				return x.address;
+			});
+
+			let version = this.version * 1;
+			let threshold = this.threshold * 1;
+
+			const multisigParams = {
+				version: version,
+				threshold: threshold,
+				addrs: addressList,
+			};
+			let multisigaddr = algosdk.multisigAddress(multisigParams);
+			if (fromAddr !== multisigaddr) {
+				openErrorNotificationWithIcon(WRONG_ADDRESS, WRONG_ADDRESSES);
+			}
+			return multisigParams;
+		},
 		async sign() {
 			let txnBase64 = "";
 			txnBase64 = this.unsignedInput;
 			try {
-				switch (this.walletStore.walletKind) {
-					case WalletType.MY_ALGO: {
-						openErrorNotificationWithIcon(NOT_SUPPORT_WALLET);
-						break;
-					}
-					case WalletType.ALGOSIGNER: {
-						let signAlgoSigner = this.walletStore.webMode as WebMode;
-
-						let sender = algosdk.decodeUnsignedTransaction(
-							convertBase64ToUnit8Array(txnBase64)
-						);
-						let fromAddr = algosdk.encodeAddress(sender.from.publicKey);
-						let addr = this.addresses.map((x) => {
-							return x.address;
-						});
-
-						let version = this.version * 1;
-						let threshold = this.threshold * 1;
-
-						const multisigParams = {
-							version: version,
-							threshold: threshold,
-							addrs: addr,
-						};
-						let multisigaddr = algosdk.multisigAddress(multisigParams);
-						if (fromAddr != multisigaddr) {
-							openErrorNotificationWithIcon(WRONG_ADDRESS, WRONG_ADDRESSES);
+				if (this.walletStore.walletKind) {
```

Comment:
should we need an `if` here since we already have switch case default

Reply:
we add the switch afterwards, to throw the error we have the if case, in else we throw the inactive wallet error